romeo/src/state.rs

```diff
@@ -232,10 +238,23 @@ fn process_bitcoin_block(
 	bitcoin_height: u32,
 	block: Block,
 ) -> (State, Vec<Task>) {
```

Comment:
Question: why does block processing depend on the config? Shouldn't the peg wallet be derived from something on-chain? I find odd filtering deposits and withdrawals based on an immutable config field.

Reply:
The romeo engine is the single source of truth and the on chain info is only a copy of it for information.

Romeo has no (and will not have) function to change the sbtc wallet address while running.